🐞 Go codegen panics when the go directive is missing from go.mod

### What is the issue?

Although technically it's not easy to get to unless you accidentally remove the go directive from go.mod file, it's possible.

https://github.com/dagger/dagger/blob/3cb2e55d898f3cae4673188c46107b76090db62f/cmd/codegen/generator/go/generator.go#L216


### Dagger version

dagger v0.15.1 (registry.dagger.io/engine:v0.15.1) darwin/arm64

### Steps to reproduce

_No response_

### Log output

```
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                                                                                                                                                                                                                                                                          
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7aa930]                                                                                                                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
goroutine 1 [running]:                                                                                                                                                                                                                                                                                                                                                                                                                                                           
github.com/dagger/dagger/cmd/codegen/generator/go.(*GoGenerator).bootstrapMod(0x40003b60e0, {0xb72830?, 0x40003ae090?}, 0x40003b4690)                                                                                                                                                                                                                                                                                                                                            
        /app/cmd/codegen/generator/go/generator.go:216 +0x490                                                                                                                                                                                                                                                                                                                                                                                                                    
github.com/dagger/dagger/cmd/codegen/generator/go.(*GoGenerator).Generate(0x40003b60e0, {0xb72830, 0x40003ae090}, 0x40003ce0f0, {0x40000bf559, 0x7})                                                                                                                                                                                                                                                                                                                             
        /app/cmd/codegen/generator/go/generator.go:88 +0x550                                                                                                                                                                                                                                                                                                                                                                                                                     
main.generate({0xb72830, 0x40003ae090}, 0x40003ce0f0, {0x40000bf559, 0x7}, {{0x9bbbf2, 0x2}, {0xffffd5ba3cad, 0x4}, {0xffffd5ba3ce3, ...}, ...})                                                                                                                                                                                                                                                                                                                                 
        /app/cmd/codegen/codegen.go:105 +0x1a0                                                                                                                                                                                                                                                                                                                                                                                                                                   
main.Generate({0xb72830, 0x40003ae090}, {{0x9bbbf2, 0x2}, {0xffffd5ba3cad, 0x4}, {0xffffd5ba3ce3, 0x7}, {0xffffd5ba3ccd, 0x7}, ...})                                                                                                                                                                                                                                                                                                                                             
        /app/cmd/codegen/codegen.go:48 +0x2ac                                                                                                                                                                                                                                                                                                                                                                                                                                    
main.ClientGen(0x1190420, {0x9bc440?, 0x4?, 0x9bc388?})                                                                                                                                                                                                                                                                                                                                                                                                                          
        /app/cmd/codegen/main.go:106 +0x2a0                                                                                                                                                                                                                                                                                                                                                                                                                                      
github.com/spf13/cobra.(*Command).execute(0x1190420, {0x400013e010, 0x8, 0x8})                                                                                                                                                                                                                                                                                                                                                                                                   
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x834                                                                                                                                                                                                                                                                                                                                                                                                          
github.com/spf13/cobra.(*Command).ExecuteC(0x1190420)                                                                                                                                                                                                                                                                                                                                                                                                                            
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344                                                                                                                                                                                                                                                                                                                                                                                                         
github.com/spf13/cobra.(*Command).Execute(...)                                                                                                                                                                                                                                                                                                                                                                                                                                   
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041                                                                                                                                                                                                                                                                                                                                                                                                                
main.main()                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
        /app/cmd/codegen/main.go:139 +0x24                                                                                                                                                                                                                                                                                                                                                                                                                                       
! process "codegen --output /src --module-context-path /src/aws-cli --module-name aws-cli --introspection-json-path /schema.json" did not complete successfully: exit code: 2
Error: failed to get module SDK: input: moduleSource.withContextDirectory.asModule failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to get modified source directory for go module sdk codegen: select: process "codegen --output /src --module-context-path /src/aws-cli --module-name aws-cli --introspection-json-path /schema.json" did not complete successfully: exit code: 2
```

--- a/cmd/codegen/generator/go/generator.go
+++ b/cmd/codegen/generator/go/generator.go
@@ -213,6 +213,9 @@ func (g *GoGenerator) bootstrapMod(ctx context.Context, mfs *memfs.FS) (*Package
 	//
 	// if this fails, then the go.mod version is too high! and in that case, we
 	// won't be able to load the resulting package
+	if mod.Go == nil {
+		return nil, false, fmt.Errorf("go.mod has no go directive")
+	}
 	if semver.Compare("v"+mod.Go.Version, "v"+goVersion) > 0 {
 		return nil, false, fmt.Errorf("existing go.mod has unsupported version %v (highest supported version is %v)", mod.Go.Version, goVersion)
 	}

